Add tests for `Account` method `apply_delta()`

phklive opened the pull request:
In this PR I propose an addition of tests for the method `apply_delta` of the `Account` struct. 

This will enable us to verify that the delta updates are correctly applied and that the errors are caught.

closes: #417

phklive commented:
@bobbinth do you believe that I should add modifications to the account_storage too? or only manipulating the vault and nonce should be okay?

bobbinth reviewed (commented):
Looks good! Thank you!

> @bobbinth do you believe that I should add modifications to the account_storage too? or only manipulating the vault and nonce should be okay?

If not too difficult, I think that would be good to have as well.

bobbinth commented:
This PR does not yet test updating of storage, right? I see a storage slot being initialized but not updated - or am I missing something?

phklive commented:
> This PR does not yet test updating of storage, right? I see a storage slot being initialized but not updated - or am I missing something?

Fixed it in the latest commit, added storage manipulation. Only added storage init on first iteration.

bobbinth reviewed (approved):
Looks good! Thank you! Let's create an issue to extend the tests to handle the storage part more generically. Could be a "good first issue" for someone.

phklive commented:
> Looks good! Thank you! Let's create an issue to extend the tests to handle the storage part more generically. Could be a "good first issue" for someone.

Absolutely. The issue has been opened here: https://github.com/0xPolygonMiden/miden-base/issues/444

Could you be slightly clearer of what I could have improved to make those tests better / what is missing? 

Thank you.

bobbinth commented:
I think the main thing is to not hard-code account storage changes inside `build_account_delta()` but pass them in as arguments (similar to what we have for the vault changes).
